urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/io/fileio/htaccess/HypertextAccessFile.java

```diff
@@ -0,0 +1,1052 @@
+	this.addedHandlers       = new TreeMap<String,List<String>>( CaseInsensitiveComparator.sharedInstance );
+	this.addedCharsets       = new TreeMap<String,String>( CaseInsensitiveComparator.sharedInstance );
+	this.directoryIndexList  = new ArrayList<String>(1);
+	this.errorDocumentMap    = new TreeMap<Integer,String>();
+    }
+
+
+    /**
+     * Reads the htaccess configuration from the given file.
+     *
+     * @param file        The input file (must be a text file in .htaccess format).
+     * @param strictMode  If this mode is set to true the reader will NOT ignore any unknown/unexpected 
+     *                    irregularities but throw a ParseException. There are even some un-handled keywords
+     *                    which still lack some implementation (set to false if you want to ignore them).
+     * @throws IOException    If any IO errors occur.
+     * @throws ParseException If the passed file cannot be parsed or is not a htaccess file.
+     **/
+    public static HypertextAccessFile read( File file,
+					    boolean strictMode )
+	throws IOException,
+	       ParseException {
+
+	
+	FileInputStream in = new FileInputStream(file);
+	HypertextAccessFile htaccess = read( in,
+					     strictMode 
+					     );
+	htaccess.sourceFile = file;
+	in.close();
+	return htaccess;
+    }    
+
+
+    /**
+     * Reads the htaccess configuration from the given inputstream.
+     *
+     * @param in The input stream (must deliver a text file in .htaccess format).
+     * @param strictMode  If this mode is set to true the reader will NOT ignore any unknown/unexpected 
+     *                    irregularities but throw a ParseException. There are even some un-handled keywords
+     *                    which still lack some implementation (set to false if you want to ignore them).
+     * @throws IOException    If any IO errors occur.
+     * @throws ParseException If the passed data cannot be parsed or is not a htaccess file.
+     **/
+    public static HypertextAccessFile read( InputStream in,
+					    boolean strictMode
+					    )
+	throws IOException,
+	       ParseException {
+
+	
+	return read( new LineNumberReader( new InputStreamReader(in) ),
+		     strictMode
+		     );
+    }    
+
+    /**
+     * Reads the htaccess configuration from the given line number reader.
+     *
+     * @param lnr The line number reader to read the data from (must deliver a text file in .htaccess format).
+     * @param strictMode  If this mode is set to true the reader will NOT ignore any unknown/unexpected 
+     *                    irregularities but throw a ParseException. There are even some un-handled keywords
+     *                    which still lack some implementation (set to false if you want to ignore them).
+     * @throws IOException    If any IO errors occur.
+     * @throws ParseException If the passed data cannot be parsed or is not a htaccess file.
+     **/
+    public static HypertextAccessFile read( LineNumberReader lnr,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.